addons/web/models/models.py

```diff
@@ -212,34 +212,97 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
         return r
 
     @api.model
-    def search_panel_select_range(self, field_name):
+    def search_panel_select_range(self, field_name, **kwargs):
         """
-        Return possible values of the field field_name (case select="one")
-        and the parent field (if any) used to hierarchize them.
+        Return possible values of the field field_name (case select="one"),
+        possibly with counters and the parent field (if any) used to hierarchize them.
 
-        :param field_name: the name of a many2one category field
+        :param field_name: the name of a field;
+            of type many2one or selection.
+        :param search_domain: base domain of search
+        :param disable_counters: whether to count records by value
         :return: {
             'parent_field': parent field on the comodel of field, or False
             'values': array of dictionaries containing some info on the records
                         available on the comodel of the field 'field_name'.
-                        The display name (and possibly parent_field) are fetched.
+                        The display name, the count (how many records with that value)
+                        and possibly parent_field are fetched.
         }
         """
         field = self._fields[field_name]
-        supported_types = ['many2one']
+        supported_types = ['many2one', 'selection']
         if field.type not in supported_types:
             raise UserError(_('Only types %(supported_types)s are supported for category (found type %(field_type)s)') % ({
                             'supported_types': supported_types, 'field_type': field.type}))
 
-        Comodel = self.env[field.comodel_name]
-        fields = ['display_name']
-        parent_name = Comodel._parent_name if Comodel._parent_name in Comodel._fields else False
-        if parent_name:
-            fields.append(parent_name)
-        return {
-            'parent_field': parent_name,
-            'values': Comodel.with_context(hierarchical_naming=False).search_read([], fields),
-        }
+        search_domain = kwargs.get('search_domain', [])
+        disable_counters = kwargs.get('disable_counters', False)
+
+        category_values = []
+
+        # retrieve all possible values, and return them with their label and counter
+        if field.type == 'many2one':
+            def group_id_name(value):
+                return value or (False, _("Not Set"))
+
+            # get counter
+            counters = {}
+            if not disable_counters:
```

Comment:
This looks a lot like copy-paste from method `search_panel_select_multi_range`.
Isn't this method just a special case of `search_panel_select_multi_range`?

Reply:
There are just small differences between select one and select multi case

Reply:
will be done later on in another PR (I don't want to have to undo what would be done here)